cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -241,10 +241,10 @@
      */
     private KafkaCluster(Reconciliation reconciliation, HasMetadata resource) {
         super(reconciliation, resource, APPLICATION_NAME);
-        this.name = kafkaClusterName(cluster);
-        this.serviceName = serviceName(cluster);
-        this.headlessServiceName = headlessServiceName(cluster);
-        this.ancillaryConfigMapName = metricAndLogConfigsName(cluster);
+        this.name = KafkaResources.kafkaStatefulSetName(cluster);
```

Comment:
This `name` IIRC is used for the `StrimziPodSet` as well (of course for the `StatefulSet` now). Should we try to change the method name `kafkaStatefulSetName` with something more generic (which apply on both "sets")? It's actually the `kafkaClusterName` or? Maybe we can reuse this method name in the `KafkaResources` class as you removed the method in this class.

Reply:
I think this is complicated - it is used to name many things (although I actually expect the StrimziPodSets to use this only temporarily and not long term). However, there is nothing like a `kafkaClusterName` in reality. The _Kafka cluster_ is pretty much the whole Kafka CR and this is not its name. So that name would be quite confusing.

So I guess the question is whether we have any better name ... and whether it really belongs to this PR since that is really part of the `KafkaResources` class and not `Kafkacluster` class anyway.